blackrock.com: REGRESSION(283494@main): Timeouts observed with TLS upgrade

RupinMittal opened the pull request:
#### cde2625ebe14befd13b4a825f77b68645b5ddb12
<pre>
blackrock.com: REGRESSION(283494@main): Timeouts observed with TLS upgrade
<a href="https://bugs.webkit.org/show_bug.cgi?id=293286">https://bugs.webkit.org/show_bug.cgi?id=293286</a>
<a href="https://rdar.apple.com/151910984">rdar://151910984</a>

Reviewed by Matthew Finkel.

An HTTP request may be optimistically upgraded to an HTTPS request
when possible. The HTTPS request will be a redirect request from
the original HTTP request. In this event, the original HTTP request
will get a timeout interval of 3 seconds. This timeout currently
gets carried over to the redirect request. In cases where the
redirect is not completed within the 3 seconds, the redirect request
will timeout. This results in the site hanging and a terrible
customer experience.

We don&apos;t need to have the short timeout on the redirect request.

To fix this, we ensure that in cases where the request is scheme
upgraded, the redirect does not get the 3 second timeout. Instead,
it&apos;s timeout will be set at the platform&apos;s default value (this is
usually a large value--for example INT_MAX on iOS).

This change was causing a layout test to fail
(http/tests/xmlhttprequest/on-network-timeout-error-during-preflight.html):

The issue was two things:

1. Layout tests were implicitly relying on a different part of the
   code specifying a timeout value of 60 seconds. But with this change,
   the redirect request case will no longer implicitly rely on that,
   but rather explicity use ResourceRequestBase::defaultTimeoutInterval(),
   which is 0 for non-iOS platforms. From testing, it seems like this is
   interpreted as INT_MAX (or at least some high non-zero value) by
   lower level frameworks. So there is essentially no timeout. And this
   test is checking for a timeout. So we change TestController to explicity
   use 60 seconds as the default timeout for layout tests. We do the same
   for DumpRenderTree for WebKit1.

2. Even once we fixed this, the default value wasn&apos;t truly being
   updated to 60 seconds. The issue was that the WebKit C API to set
   the default value propagated the new value to all Web Processes
   but not to Network Processes. In this test, the value is accessed
   from the Network Process. Since the 60 seconds value wasn&apos;t set for
   Network Processes, it returned 0 seconds, which gets interpreted as
   INT_MAX (or some other higher non-zero value), and thus the test
   failed. We fix this by propagating the new default value from the
   API to Network Processes as well.

So we now explicitly set the default value to 60 seconds for each
layout test. And the WKURLRequestSetDefaultTimeoutInterval API
will propagate the value to Network Processes as well as Web Processes.

* LayoutTests/ipc/serialized-type-info.html:
* Source/WTF/wtf/Markable.h:
(WTF::DoubleMarkableTraits::isEmptyValue):
(WTF::DoubleMarkableTraits::emptyValue):
* Source/WebCore/animation/WebAnimationTypes.h:
(WebCore::WebAnimationsMarkableDoubleTraits::isEmptyValue): Deleted.
(WebCore::WebAnimationsMarkableDoubleTraits::emptyValue): Deleted.
* Source/WebCore/platform/network/ResourceRequestBase.cpp:
(WebCore::ResourceRequestBase::resetTimeoutInterval):

This will reset the timeout to the default value.

* Source/WebCore/platform/network/ResourceRequestBase.h:
* Source/WebCore/platform/network/cocoa/ResourceRequestCocoa.mm:
(WebCore::ResourceRequest::doUpdatePlatformRequest):

This function updates the NSRequest with the current information
stored in the updated WebCore::ResourceRequest.

Before the update happens, in the case of the redirect, the NSRequest
will contain the timeout from the original request (the short
time interval). Currently, if the updated WebCore::ResourceRequest&apos;s
timeout interval is 0, we don&apos;t change the NSRequest&apos;s timeout, so
it remains as the short timeout.

We update this so that if the ResourceRequest&apos;s timeout is
0, we&apos;ll use the ResourceRequest&apos;s default timeout (generally a large
value, like INT_MAX on iOS).

* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::initializeNetworkProcess):
(WebKit::NetworkProcess::setDefaultRequestTimeoutInterval):
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/NetworkProcess/NetworkProcess.messages.in:
* Source/WebKit/NetworkProcess/NetworkProcessCreationParameters.h:

The default timeout for Network Processes can be set here.

* Source/WebKit/NetworkProcess/NetworkProcessCreationParameters.serialization.in:
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::NetworkResourceLoader::willSendRedirectedRequestInternal):

If the original request was scheme upgraded, set the timeout on the
redirect request to the default value. If it&apos;s set it to 0, then when this
request is converted to an NSRequest, the default timeout will be used.

* Source/WebKit/Platform/IPC/ArgumentCoders.h:
* Source/WebKit/Shared/API/APIURLRequest.cpp:
(API::URLRequest::setDefaultTimeoutInterval):

When the default timeout is changed, this value gets propagated
to all the Web Processes. We make a change here to make sure the
value is propagated to all Network Processes as well (otherwise
the two are inconsistent).

* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::sendCreationParametersToNewProcess):
(WebKit::NetworkProcessProxy::setDefaultRequestTimeoutInterval):

We set the timeout for new Network Processes in exactly the same
way we do for WebProcesses.

* Source/WebKit/UIProcess/Network/NetworkProcessProxy.h:
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::initializeWebProcess):
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(runTest):

To make layout tests pass (as described above) we now explicitly
set the default value to 60 seconds for each layout test.

 Tools/TestWebKitAPI/Tests/WebKitCocoa/Navigation.mm:
(TEST(WKNavigation, DISABLED_PreferredHTTPSPolicyAutomaticHTTPFallbackRedirectNo3SecondTimeout)):

New API test to test this behavior. This functionality depends on
a change made to a lower level system framework (libnetcore). That
change currently isn&apos;t on EWS, so this test is disabled for now.
We&apos;ll enable it once EWS is upgraded.

* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::resetStateToConsistentValues):

To make layout tests pass (as described above) we now explicitly
set the default value to 60 seconds for each layout test.

Canonical link: <a href="https://commits.webkit.org/297128@main">https://commits.webkit.org/297128@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3351f6562c0a9ca906101745312d9bd4a6d637bb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110609 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/30268 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20701 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116635 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60876 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/112572 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30947 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38857 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/84119 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113557 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24715 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99600 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64559 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/24078 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17740 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60430 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/103100 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/94101 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17799 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119425 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/109163 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37649 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27964 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/93085 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/38023 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95871 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92908 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23672 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37917 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15660 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33626 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37544 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/43015 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/133438 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/37207 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36057 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40546 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38915 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/Shared/API/APIURLRequest.cpp:
```diff
@@ -49,6 +50,11 @@ void URLRequest::setDefaultTimeoutInterval(double timeoutInterval)
+
+    WebsiteDataStore::forEachWebsiteDataStore([timeoutInterval](auto& websiteDataStore) {
+        if (RefPtr networkProcessProxy = websiteDataStore.networkProcessIfExists())
+            networkProcessProxy->setDefaultRequestTimeoutInterval(timeoutInterval);
```
On Cocoa ports, we use a singleton NetworkProcessProxy so all data stores would end up with the same network process proxy. In this case, wouldn't it keep sending IPC to the same network process?

RupinMittal replied:
Oh I see. Yeah, I suppose we would be repeatedly sending IPC to the same network process. 

I just updated the PR with this -- maintain a list of all the Network Processes that have already been sent a message and only send IPC to a Network Process that hasn't been sent to yet. Seems to work locally, hopefully will work on bots. 

What do you think?

cdumez replied:
Wouldn't it be simpler to use `NetworkProcessProxy::allNetworkProcesses()`?

RupinMittal replied:
It would! I didn't know that existed. Just updated the patch to use this.

cdumez commented on Source/WebKit/NetworkProcess/NetworkProcessCreationParameters.h:
```diff
@@ -82,6 +82,8 @@ struct NetworkProcessCreationParameters {
     HashSet<String> localhostAliasesForTesting;
     Vector<WebCore::OrganizationStorageAccessPromptQuirk> storageAccessPromptQuirksData;
+
+    double defaultRequestTimeoutInterval { INT_MAX };
```
We should not use special values like this in modern code. use std::optional or WTF::Markable. WTF::Markable is our modern way of doing special values. std::optional is good too but will take more space since it does not rely on a special value.

RupinMittal replied:
That does make sense. But if we use Markable, and the client doesn't manually set a default value, then the Network Process won't have one. And it will be inconsistent with the WebProcess.

I was simply trying to match what we do for the WebProcess. The code in the Web Process is identical to this. My goal was to make sure that the WebProcess and NetworkProcess are in sync with each other. Setting INT_MAX does make sense as a default.

Is that fine or should I still change the Network Process to use Markable?

cdumez replied:
We should consistently use Markable I think but you can feel free to do that in a follow up IF this patch will get cherry-picked to a SU branch (didn't check).

RupinMittal replied:
Got it. Just updated the patch to use Markable for the both the Network Process and Web Process.

cdumez commented on Tools/WebKitTestRunner/TestController.cpp:
```diff
@@ -1389,6 +1389,8 @@ bool TestController::resetStateToConsistentValues(const TestOptions& options, Re
 
     WKWebsiteDataStoreResetResourceMonitorThrottler(websiteDataStore(), nullptr, nullptr);
 
+    WKURLRequestSetDefaultTimeoutInterval(Seconds { 60 }.value());
```
`60_s.value()`?

RupinMittal replied:
Fixed!

cdumez commented on Tools/DumpRenderTree/mac/DumpRenderTree.mm:
```diff
@@ -1986,6 +1987,8 @@ static void runTest(const std::string& inputLine)
     gTestRunner->clearAllDatabases();
     gTestRunner->clearNotificationPermissionState();
 
+    WKURLRequestSetDefaultTimeoutInterval(Seconds { 60 }.value());
```
`60_s.value()`?

RupinMittal replied:
Fixed!

cdumez commented on Source/WebKit/NetworkProcess/NetworkProcessCreationParameters.h:
```diff
@@ -42,6 +43,18 @@
 
 namespace WebKit {
 
+struct NetworkProcessCreationParametersMarkableDoubleTraits {
```
Can we define this in Markable.h, next to the Traits for `float` and get rid of `WebAnimationsMarkableDoubleTraits` in the process?

RupinMittal replied:
I've moved the traits to Markable.h. Seems like we can't eliminate the second traits parameter without more involved changes for now. But as discussed, I've edited the test to allow this to work for now.

sysrqb commented on Tools/DumpRenderTree/mac/DumpRenderTree.mm:
```diff
@@ -1986,6 +1988,8 @@ static void runTest(const std::string& inputLine)
     gTestRunner->clearAllDatabases();
     gTestRunner->clearNotificationPermissionState();
 
+    WKURLRequestSetDefaultTimeoutInterval((60_s).value());
```
Do we need the parenthesis around 60_s?

RupinMittal replied:
Unfortunately, yes. If we don't, the compiler thinks that `_s` and `.value` are a single operator:

```
/OpenSource/Tools/TestWebKitAPI/Tests/WebKitCocoa/Navigation.mm:2641:45: 
error: no matching literal operator for call to 'operator""_s.value' with argument of 
type 'unsigned long long' or 'const char *', and no matching literal operator template
2641 |     WKURLRequestSetDefaultTimeoutInterval(60_s.value());
```

sysrqb replied:
Yeah, looks like we need either the parentheses or we can use a space: https://en.cppreference.com/w/cpp/language/user_literal.html#:~:text=Same%20applies%20to%20dot%20operator%20following%20an%20integer%20or%20floating%2Dpoint%20user%2Ddefined%20literal:

```
auto a = 4s.count();   // Error
auto b = 4s .count();  // OK
auto c = (4s).count(); // OK
```

RupinMittal replied:
Ah interesting. Yeah, in that case I prefer the parentheses over the space.

sysrqb commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/Navigation.mm:
```diff
@@ -2633,6 +2634,79 @@ HTTPServer httpServer({
+{
+    using namespace TestWebKitAPI;
+
+    WKURLRequestSetDefaultTimeoutInterval(Seconds { 60 }.value());
```
`_s` change here too

RupinMittal replied:
Fixed!

sysrqb commented on Tools/WebKitTestRunner/TestController.cpp:
```diff
@@ -1389,6 +1389,8 @@ bool TestController::resetStateToConsistentValues(const TestOptions& options, Re
 
     WKWebsiteDataStoreResetResourceMonitorThrottler(websiteDataStore(), nullptr, nullptr);
 
+    WKURLRequestSetDefaultTimeoutInterval((60_s).value());
```
parentheses?

RupinMittal replied:
Same as above.

sysrqb commented on Source/WebKit/UIProcess/Network/NetworkProcessProxy.h:
```diff
@@ -348,6 +348,8 @@ class NetworkProcessProxy final : public AuxiliaryProcessProxy {
     void resetResourceMonitorThrottlerForTesting(PAL::SessionID, CompletionHandler<void()>&&);
 #endif
 
+    void setDefaultRequestTimeoutInterval(double timeoutInterval);
```
nit: We probably don't need the argument name here because it's relatively clear what the double represents, but feel free to leave it if you prefer.

RupinMittal replied:
Fixed!

sysrqb reviewed (approved):
I think this looks good! For the title, you can use `REGRESSION(283494@main)`.

RupinMittal committed:
blackrock.com: REGRESSION(283494@main): Timeouts observed with TLS upgrade
https://bugs.webkit.org/show_bug.cgi?id=293286
rdar://151910984

Reviewed by Matthew Finkel.

An HTTP request may be optimistically upgraded to an HTTPS request
when possible. The HTTPS request will be a redirect request from
the original HTTP request. In this event, the original HTTP request
will get a timeout interval of 3 seconds. This timeout currently
gets carried over to the redirect request. In cases where the
redirect is not completed within the 3 seconds, the redirect request
will timeout. This results in the site hanging and a terrible
customer experience.

We don't need to have the short timeout on the redirect request.

To fix this, we ensure that in cases where the request is scheme
upgraded, the redirect does not get the 3 second timeout. Instead,
it's timeout will be set at the platform's default value (this is
usually a large value--for example INT_MAX on iOS).

This change was causing a layout test to fail
(http/tests/xmlhttprequest/on-network-timeout-error-during-preflight.html):

The issue was two things:

1. Layout tests were implicitly relying on a different part of the
   code specifying a timeout value of 60 seconds. But with this change,
   the redirect request case will no longer implicitly rely on that,
   but rather explicity use ResourceRequestBase::defaultTimeoutInterval(),
   which is 0 for non-iOS platforms. From testing, it seems like this is
   interpreted as INT_MAX (or at least some high non-zero value) by
   lower level frameworks. So there is essentially no timeout. And this
   test is checking for a timeout. So we change TestController to explicity
   use 60 seconds as the default timeout for layout tests. We do the same
   for DumpRenderTree for WebKit1.

2. Even once we fixed this, the default value wasn't truly being
   updated to 60 seconds. The issue was that the WebKit C API to set
   the default value propagated the new value to all Web Processes
   but not to Network Processes. In this test, the value is accessed
   from the Network Process. Since the 60 seconds value wasn't set for
   Network Processes, it returned 0 seconds, which gets interpreted as
   INT_MAX (or some other higher non-zero value), and thus the test
   failed. We fix this by propagating the new default value from the
   API to Network Processes as well.

So we now explicitly set the default value to 60 seconds for each
layout test. And the WKURLRequestSetDefaultTimeoutInterval API
will propagate the value to Network Processes as well as Web Processes.

* LayoutTests/ipc/serialized-type-info.html:
* Source/WTF/wtf/Markable.h:
(WTF::DoubleMarkableTraits::isEmptyValue):
(WTF::DoubleMarkableTraits::emptyValue):
* Source/WebCore/animation/WebAnimationTypes.h:
(WebCore::WebAnimationsMarkableDoubleTraits::isEmptyValue): Deleted.
(WebCore::WebAnimationsMarkableDoubleTraits::emptyValue): Deleted.
* Source/WebCore/platform/network/ResourceRequestBase.cpp:
(WebCore::ResourceRequestBase::resetTimeoutInterval):

This will reset the timeout to the default value.

* Source/WebCore/platform/network/ResourceRequestBase.h:
* Source/WebCore/platform/network/cocoa/ResourceRequestCocoa.mm:
(WebCore::ResourceRequest::doUpdatePlatformRequest):

This function updates the NSRequest with the current information
stored in the updated WebCore::ResourceRequest.

Before the update happens, in the case of the redirect, the NSRequest
will contain the timeout from the original request (the short
time interval). Currently, if the updated WebCore::ResourceRequest's
timeout interval is 0, we don't change the NSRequest's timeout, so
it remains as the short timeout.

We update this so that if the ResourceRequest's timeout is
0, we'll use the ResourceRequest's default timeout (generally a large
value, like INT_MAX on iOS).

* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::initializeNetworkProcess):
(WebKit::NetworkProcess::setDefaultRequestTimeoutInterval):
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/NetworkProcess/NetworkProcess.messages.in:
* Source/WebKit/NetworkProcess/NetworkProcessCreationParameters.h:

The default timeout for Network Processes can be set here.

* Source/WebKit/NetworkProcess/NetworkProcessCreationParameters.serialization.in:
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
(WebKit::NetworkResourceLoader::willSendRedirectedRequestInternal):

If the original request was scheme upgraded, set the timeout on the
redirect request to the default value. If it's set it to 0, then when this
request is converted to an NSRequest, the default timeout will be used.

* Source/WebKit/Platform/IPC/ArgumentCoders.h:
* Source/WebKit/Shared/API/APIURLRequest.cpp:
(API::URLRequest::setDefaultTimeoutInterval):

When the default timeout is changed, this value gets propagated
to all the Web Processes. We make a change here to make sure the
value is propagated to all Network Processes as well (otherwise
the two are inconsistent).

* Source/WebKit/Shared/WebProcessCreationParameters.h:
* Source/WebKit/Shared/WebProcessCreationParameters.serialization.in:
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::sendCreationParametersToNewProcess):
(WebKit::NetworkProcessProxy::setDefaultRequestTimeoutInterval):

We set the timeout for new Network Processes in exactly the same
way we do for WebProcesses.

* Source/WebKit/UIProcess/Network/NetworkProcessProxy.h:
* Source/WebKit/WebProcess/WebProcess.cpp:
(WebKit::WebProcess::initializeWebProcess):
* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(runTest):

To make layout tests pass (as described above) we now explicitly
set the default value to 60 seconds for each layout test.

 Tools/TestWebKitAPI/Tests/WebKitCocoa/Navigation.mm:
(TEST(WKNavigation, DISABLED_PreferredHTTPSPolicyAutomaticHTTPFallbackRedirectNo3SecondTimeout)):

New API test to test this behavior. This functionality depends on
a change made to a lower level system framework (libnetcore). That
change currently isn't on EWS, so this test is disabled for now.
We'll enable it once EWS is upgraded.

* Tools/WebKitTestRunner/TestController.cpp:
(WTR::TestController::resetStateToConsistentValues):

To make layout tests pass (as described above) we now explicitly
set the default value to 60 seconds for each layout test.

Canonical link: https://commits.webkit.org/297128@main
